src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionVerticle.java

```diff
@@ -106,13 +116,16 @@ public ImageRegionVerticle(
             LutProvider lutProvider,
             int maxTileLength,
             ZarrPixelsService pixelsService,
-            IScale iScale)
+            IScale iScale,
+            String omeroDataDir)
     {
         this.compressionService = compressionService;
         this.lutProvider = lutProvider;
         this.maxTileLength = maxTileLength;
         this.pixelsService = pixelsService;
         this.iScale = iScale;
+        this.omeroDataDir = omeroDataDir;
+        this.ioService = new OriginalFilesService(omeroDataDir, true);
```

Comment:
If you're going to use this maybe inject it like we do in the server by cribbing or using the existing Spring bean?

See:

* https://github.com/ome/omero-server/blob/master/src/main/resources/ome/services/service-ome.io.nio.OriginalFilesService.xml

Once you do that you can ask `getFilesPath()` just from the OriginalFile ID once checking `canRead()`.